### PR TITLE
Initial BLE scanning

### PR DESCRIPTION
**Pull request by drewmiley**
Resolves: #20

- [x] Assigned a Reviewer to the **PR**.
- [x] Assigned a Project to the **PR**.
- [ ] Added **tests** for changed code.

## What Changed and Why?

  1. Added https://github.com/innoveit/react-native-ble-manager sample demo
  1. Migrated code to React Hooks style
  1. Located this within Devices tab
  1. Pulled out two rendering components

## How to Review Changes?

  1. Run app and test it works on Samsung Galaxy A40 (I'm running it on a Sony Xperia X2)
  1. Review code style and see if acceptable

### CODE REVIEW

**Comment by jawrainey** [comment]
@drewmiley -- great work! The BLE connectivity works: I can scan and view nearby BLE devices, connect/disconnect to a BLE device. I tested on both a Samsung A40 and OnePlus7. However, there's a couple of issues:

- **Connecting/Disconnecting UI state**: when I tap a device to connect to it, the console logs that this was a success yet the UI does not update, i.e. the device does not turn green. If I then tap "retrieve connected peripherals" again the list is correct (i.e. my device is connected). Again no UI update has been made. However, when I tap "scan bluetooth" the associated device is changed to green to indicate that it is connected. *The same occurs when disconnecting from a device*

- **UI state**: when I connect to a device and then turn the device off, if I rescan for BLE devices the device I just turned off is also displayed in the list. Tapping the item disconnects it indicating it's saved in state. *Note:* this only occurs if you disconnect the device while remaining on the Bluetooth scanner screen; if navigate back and return to this screen the device is not listed (related to mount/unmount?)

- **Warning**: when viewing the Bluetooth scanner page: ` WARN     VirtualizedLists should never be nested inside plain ScrollViews with the same orientation - use another VirtualizedList-backed container instead. `

File: src/containers/Ble.js [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+    let newperipherals = peripherals;
+    for (let i = 0; i < results.length; i++) {
+      let peripheral = results[i];
+      peripheral.connected = true;
+      newperipherals.set(peripheral.id, peripheral);
+      setPeripherals(newperipherals);
+    }
+  };
+
+  const test = async peripheral => {
```
Comment (jawrainey) [inline src/containers/Ble.js:173]: I'd rename this to something more meaningful -- maybe "device"?

File: src/containers/Ble.js [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+    backgroundColor: '#FFF',
+    width: window.width,
+    height: window.height,
+  },
+  scroll: {
+    flex: 1,
+    backgroundColor: '#f0f0f0',
+    margin: 10,
+  },
+  NoPeripherals: {
```
Comment (jawrainey) [inline src/containers/Ble.js:241]: Should start as lowercase

File: src/containers/Ble.js (hunk 1 above)
Comment (jawrainey) [inline src/containers/Ble.js:173]: Could be renamed to be more meaningful, e.g. "device"?

Comment (drewmiley) [reply src/containers/Ble.js:172]: Happy to change this - I guess we would want peripherals -> devices everywhere though?

**Review by farzanehfar — APPROVED** [review]
Approved because:

* The logic have taken from "ble_investigation" branch that developed by @farzanehfar 
* No Logic has changed.
* This branch is a structure refactor from class based to Hub.
* Still works for onePlus6 and Galaxy 9 (Similar as ble_investigation branch) (Two current available devices)

**Pull request merged**